presto-orc/src/main/java/com/facebook/presto/orc/stream/PresentOutputStream.java

```diff
@@ -101,23 +101,21 @@ public void close()
         return Optional.of(booleanOutputStream.getCheckpoints());
     }
 
-    public long getDataStreamBytes()
-    {
-        checkArgument(closed);
-        if (booleanOutputStream == null) {
-            return 0;
-        }
-        return booleanOutputStream.getDataStreamBytes();
-    }
-
-    public Optional<Stream> writeDataStreams(int column, SliceOutput outputStream)
+    public Optional<OutputDataStream> getOutputDataStream(int column)
     {
         checkArgument(closed);
         if (booleanOutputStream == null) {
             return Optional.empty();
         }
-        Stream stream = booleanOutputStream.writeDataStreams(column, outputStream).get();
-        return Optional.of(new Stream(stream.getColumn(), PRESENT, stream.getLength(), stream.isUseVInts()));
+        OutputDataStream outputDataStream = booleanOutputStream.getOutputDataStream(column);
+        Stream stream = new Stream(column, PRESENT, toIntExact(outputDataStream.getSizeInBytes()), outputDataStream.getStream().isUseVInts());
+        return Optional.of(new OutputDataStream(
```

Comment:
Would it make sense to refactor as a static method in `OutputDataStream`, such as `createDelegateOutputDataStream` ?

Reply:
This is the only place this happens.  I added a comment explaining what that the code is rewriting the DATA stream to a PRESENT stream, which is what PresentOutputStream needs to produce